### PR TITLE
Allow user to expand cropmodel boxes for predict_tile

### PR DESCRIPTION
**Pull request by bw4sz**
While working on the BOEM pipeline I noticed a strong performance difference between how annotators make detections and the deepforest detection models, leading to weaker cropmodel classification performance. This PR allows the user to specify in config to expand the detection both by x pixels on all sides to capture a wider extent. 

Agent statement: I conceived of the change and wrote the initial function. Cursor wrote the test which I then showed failed and that my fix improved. Cursor added the edge cases that the bounding box expansion goes off the side of the image.

### CODE REVIEW

File: src/deepforest/conf/schema.py [hunk 1]
```diff
@@ -107,6 +107,8 @@ class CropModelConfig:
     scheduler: SchedulerConfig = field(default_factory=SchedulerConfig)
     balance_classes: bool = False
     resize: list[int] = field(default_factory=lambda: [224, 224])
+    # Number of pixels to expand bbox crops for additional context (not padding)
```
Comment (jveitchmichaelis) [inline src/deepforest/conf/schema.py]: Drop this comment for consistency with the rest of the file

File: src/deepforest/datasets/cropmodel.py [hunk 2]
```diff
@@ -49,12 +49,23 @@ class BoundingBoxDataset(Dataset):
         transform: Optional transform function
         augmentations: Augmentation configuration
         resize: Optional list of [height, width] for resizing. Defaults to [224, 224]
+        expand: Number of pixels to expand the bounding box on all sides before
```
Comment (jveitchmichaelis) [inline src/deepforest/datasets/cropmodel.py]: Suggest:

```
expand: Number of pixels to expand bounding box windows in source image, boxes will be expanded in all directions and clipped to image extent.
```

**Review by jveitchmichaelis — CHANGES_REQUESTED** [review]
Added a few suggestions + request a test for boxes that would expand out of bounds.

**Comment by jveitchmichaelis** [comment]
Looks good to me, just needs a squash

**Review by jveitchmichaelis — APPROVED** [review]
